tests/platform_tests/api/test_psu.py

```diff
@@ -98,6 +98,9 @@ def test_get_name(self, duthosts, enum_rand_one_per_hwsku_hostname, localhost, p
             if self.skip_absent_psu(i,platform_api_conn):
                 continue
             name = psu.get_name(platform_api_conn, i)
+            if name in self.psu_skip_list:
```

Comment:
isn't the purpose of calling "if self.skip_absent_psu(i,platform_api_conn):" in line 98 achieve the same as what you have added to check the name to skip over those on the skip list?  If so, why do you still need this check?
Can you check again why the previous check is not accomplishing the same as your newly added check?

Same comment for the rest of all your changes.  they all have the call to "if self.skip_absent_psu(i,platform_api_conn):" already before your change.